imagetest/test_suites/hotattach/hot_attach_test.go

```diff
@@ -0,0 +1,86 @@
+//go:build cit
+// +build cit
+
+package hotattach
+
+import (
+	"context"
+	"fmt"
+	"os/exec"
+	"runtime"
+	"strings"
+	"testing"
+
+	"github.com/GoogleCloudPlatform/guest-test-infra/imagetest/utils"
+	"google.golang.org/api/compute/v1"
+)
+
+func getProjectAndZone() (string, string, error) {
+	var projectZoneUrl string
+	if runtime.GOOS == "windows" {
+		procStatus, err := utils.RunPowershellCmd("Invoke-RestMethod -Headers @{'Metadata-Flavor' = 'Google'} -Uri \"http://metadata.google.internal/computeMetadata/v1/instance/zone\"")
+		if err != nil {
+			return "", "", fmt.Errorf("failed to get project or zone on windows: %v", err)
+		}
+		projectZoneUrl = strings.TrimSpace(procStatus.Stdout)
+	} else {
+		projectZoneBytes, err := exec.Command("curl", "http://metadata.google.internal/computeMetadata/v1/instance/zone", "-H", "Metadata-Flavor: Google").Output()
+		projectZoneUrl = strings.TrimSpace(string(projectZoneBytes))
+		if err != nil {
+			return "", "", fmt.Errorf("failed to get project or zone on linux: %v", err)
+		}
+	}
+	// projectZoneUrl should be in the fomrat projects/$PROJECTNUMBER/zone/$ZONE, and we want to pass in just the $ZONE value to detach the disk.
+	projectZoneSlice := strings.Split(string(projectZoneUrl), "/")
+	if strings.ToLower(projectZoneSlice[0]) != "projects" || strings.ToLower(projectZoneSlice[2]) != "zones" || len(projectZoneSlice) != 4 {
+		return "", "", fmt.Errorf("returned string for vm metata was the wrong format: got %s", projectZoneUrl)
+	}
+
+	// return format is (projectNumber, instanceZone, nil)
+	return projectZoneSlice[1], projectZoneSlice[3], nil
+}
+
+// TestFileHotAttach is a test which checks that a file on a disk is usable, even after the disk was detached and reattached.
```

Comment:
The comment here says that this test checks that a file on disk is usable after reattach but this test doesn't look like it's reading or writing any files, just calling the detach & attach APIs.

Reply:
added